### PR TITLE
AWS Provider v3 update

### PR DESCRIPTION
**Pull request by yorinasub17**
- Update to be compatible with provider v3
- Update go and terratest to latest
- Update various utilities to latest version

### CODE REVIEW

File: modules/load-balancer-target-group/main.tf [hunk 1]
```diff
@@ -4,7 +4,16 @@
 # ----------------------------------------------------------------------------------------------------------------------
 
 terraform {
-  required_version = ">= 0.12"
+  # Source is required for required_providers in TF 13, but is only compatible with TF 12 starting 0.12.26.
+  required_version = ">= 0.12.26"
+
+  # Using provider v3.0 syntax for aws_alb_listener_rule condition blocks.
```
Comment (brikis98) [inline modules/load-balancer-target-group/main.tf:10]: I thought this syntax existed in 2.x too, and it was merely the old syntax was deprecated in 2.x, and now an error in 3.x? If so, then we don't need to strictly require 3.x here.

**Review by brikis98 — APPROVED** [review]
LGTM!

File: modules/load-balancer-target-group/main.tf [hunk 2]
```diff
@@ -43,13 +52,43 @@ resource "aws_alb_listener_rule" "http_path" {
...
   action {
-    target_group_arn = aws_alb_target_group.tg.arn
     type             = "forward"
+    target_group_arn = aws_alb_target_group.tg.arn
   }
 
-  condition {
-    field  = var.routing_condition.field
-    values = var.routing_condition.values
+  dynamic "condition" {
```
Comment (yorinasub17) [inline modules/load-balancer-target-group/main.tf:59]: Use the same approach as couchbase: https://github.com/gruntwork-io/terraform-aws-couchbase/pull/64

Comment (yorinasub17) [reply modules/load-balancer-target-group/main.tf:62]: Decided not to do that as this one is using a single map for the condition, as opposed to multiple in a list.

File: modules/load-balancer-target-group/main.tf (hunk 1 above)
Comment (yorinasub17) [reply modules/load-balancer-target-group/main.tf]: Ah you are right! Just ran a quick test forcing provider v2 (2e78176fb5993824d41ab345b953a2e3a8fd02d1) and it worked. I relaxed the constraint: d00c5c799d8721eeeef27850d931d73aa3894f09.

**Review by brikis98 — APPROVED** [review]
LGTM!

**Comment by yorinasub17** [comment]
Thanks for review! Merging and releasing now.

**Pull request merged**